### ISSUE TITLE
crowdin-cli upload sources hangs because "files are uploaded but not imported"

### ISSUE THREAD

**Issue by adepablor** [crowdin/crowdin-cli-v1#46]
Hi guys,

Already submitted this through support but I guess it would gurt to do it here.

When trying to upload sources for a branch via `crowdin-cli upload sources -b branch_name` the scripts hangs and we see this error in the crowdin UI. Only way to solve it is to hit import for all the files.....

<img width="1207" alt="screen shot 2016-04-20 at 1 53 41 pm" src="https://cloud.githubusercontent.com/assets/964907/14684829/c359906e-06ff-11e6-8e0e-65b06c07401e.png">

**Comment by mamantoha** [comment]
Hi, @adepablor .

Can you provide your config file w/o credentials?

**Comment by adepablor** [comment]
Yep

``` yaml
files:
  -
    source: '/web/WEB-INF/localization/en/*.properties'
    translation: '/web/WEB-INF/localization/%locale%/%original_file_name%'
    escape_quotes: 1
    ignore:
      - '/web/WEB-INF/localization/en/foreign_license_instructions.properties'
    languages_mapping:
      locale:
        'fr-CA': 'fr'
  -
    source: '/web/resources/js/locales/en.json'
    translation: '/web/resources/js/locales/%locale%.%file_extension%'
    languages_mapping:
      locale:
        'fr-CA': 'fr'
```

**Comment by adepablor** [comment]
To offer more context, this doesn't happen in a very consistent way, it looks very random. Thanks for taking a look!

Edit: I think it only happens for new branches (updates are fine). Although I cannot confirm 100%

**Comment by mamantoha** [comment]
Hmm. I can not reproduce such behaviour on my side. Maybe a problem with source files.
It will be helpful if you can provide one of the files (e.g. `push_notifications.properties` from the screenshot).
In addition, I'm interested what happens when you click on 'Import' button.

**Comment by adepablor** [comment]
Hi @mamantoha . It is just a normal property file with key/values like:

`push.owner.new_request.title=Trip request`

When I hit import it spins and then goes for the next file until there are no more files to import and then removes the warning. No idea what's going on behind the scenes.

Happy to send you a video or something over email maybe?

**Comment by mamantoha** [comment]
I do not think it will be helpful for me. Tomorrow I'll ask Crowdin support team about this issue.

Hope, they can help you.

**Comment by adepablor** [comment]
Thanks

**Comment by Andrulko** [comment]
Alberto,

Can you please tell us if the issue remains? Do you upload files over API and check the _Files_ tab via web UI simultaneously? 

In this case, that is quite expected you will see "Needs attention" near files that are currently imported by CLI. The warning will disappear when you will receive the following response in console from CLI:
Uploading source file `WEB-INF/localization/en/push_notifications.properties' - OK

Consequently, I'd recommend to check _Files_ tab in the project settings after all files will become imported to the system. Otherwise, you can try to refresh the page - if the file will be imported, the warning will disappear but you will see the same warning across the next file in the list.

**Comment by adepablor** [comment]
Hi Andriy,

> Can you please tell us if the issue remains?

Yes, the issue remains.

> Do you upload files over API and check the Files tab via web UI simultaneously?

I could see yesterday night that that was why I was seeing it. I honestly didn't know it was simultaneously cause we have an automatic system in place to upload branches and I just saw that a branch was taking a long time. I did it from the terminal and I saw exactly what you are describing. 

The thing is, our strings are taking almost 2 hours to upload, so I thought it was hosed but it wasn't, it was just taking a long time. So to sum up, yes it works, but that upload time is an issue itself in the problem. They are not that many strings and it wasn't taking so long a couple of days ago.

I was experimenting and it looks like the more branches we have, the more times it takes. Upload time seems to raise almost exponentially with number of branches, which is unsustainable for us. We went from around 5 minutes with one or two branches to 2 hours with 30. But we expect to have a lot more than 30 branches. 

Anything we can do to solve it?
